src/NHibernate/Proxy/DynamicProxy/ProxyCacheEntry.cs

```diff
@@ -24,6 +24,7 @@ public ProxyCacheEntry(System.Type baseType, System.Type[] interfaces)
 			}
 			BaseType = baseType;
 			Interfaces = interfaces ?? new System.Type[0];
+			_uniqueInterfaces = new HashSet<System.Type>(Interfaces);
```

Comment:
But there is the hashcode computation too. I am wrong thinking xor is permutable? (Will yield same result whatever the evaluation order.) I am no mure sure about that. But I can not find an exemple for which it would be false.